lib/wallet/rpc.js

```diff
@@ -1794,18 +1794,27 @@ class RPC extends RPCBase {
   }
 
   async getNames(args, help) {
-    if (help || args.length !== 0)
-      throw new RPCError(errs.MISC_ERROR, 'getnames');
+    if (help || args.length > 1)
+      throw new RPCError(errs.MISC_ERROR, 'getnames ( own )');
 
+    const valid = new Validator(args);
     const wallet = this.wallet;
     const height = this.wdb.height;
     const network = this.network;
+    const own = valid.bool(0, false);
 
     const names = await wallet.getNames();
     const items = [];
 
     for (const ns of names)
-      items.push(ns.getJSON(height, network));
+      if (own) {
+        const {hash, index} = ns.owner;
+        const coin = await wallet.getCoin(hash, index);
+        if (coin)
```

Comment:
The resulting statement is the same no matter of the if clause.

Reply:
I don't think so - if `own` is set to `true` then `ns.getJSON` is only pushed to the output array if the name owner is a coin in the wallet. If `own` is set to `false` then all `ns` objects returned by `wallet.getNames()` will be returned.

We should be careful though how long this extra DB lookup takes - it could really slow down the response if there's too many names in a wallet.